File: ejerc.diccionarios/Ejercicio 7/Ejercicio7.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class CestaCompra {
    public static void main(String[] args) {
        // Crear un HashMap para almacenar los artículos (clave) y sus precios (valor)
        Map<String, Double> cesta = new HashMap<>();

        // Crear objeto Scanner para leer datos del usuario
        Scanner sc = new Scanner(System.in);

        // Bucle para ingresar artículos y precios
        while (true) {
            // Solicitar nombre del artículo
            System.out.print("Ingrese el artículo (o 'salir' para terminar): ");
            String articulo = sc.nextLine().trim();

            // Si el usuario escribe "salir", se rompe el bucle
            if (articulo.equalsIgnoreCase("salir")) {
                break;
            }

            // Solicitar el precio del artículo
            System.out.print("Ingrese el precio de " + articulo + ": ");
            double precio = sc.nextDouble();
            sc.nextLine(); // Limpiar el buffer para evitar problemas en la siguiente lectura

            // Agregar el artículo y su precio al HashMap
            cesta.put(articulo, precio);
        }

        // Mostrar lista de la compra
        System.out.println("\nLista de la compra");
        double total = 0; // Variable para acumular el total de la compra

        // Recorrer el HashMap e imprimir cada artículo con su precio
        for (Map.Entry<String, Double> entry : cesta.entrySet()) {
            System.out.printf("%s: %.2f%n", entry.getKey(), entry.getValue());
            total += entry.getValue(); // Sumar el precio al total
        }

        // Imprimir el total de la compra
        System.out.printf("%nTotal de la compra: %.2f%n", total);

        // Cerrar el objeto Scanner para liberar recursos
        sc.close();
    }
}

File: ejerc.diccionarios/Ejercicio 7/Ejercicio7.py
# Crear un diccionario vacío para almacenar los artículos y sus precios
cesta = {}

# Bucle infinito para permitir al usuario ingresar varios artículos
while True:
    # Pedir el nombre del artículo
    articulo = input("Ingrese el artículo (o 'salir' para terminar): ").strip()
    
    # Si el usuario escribe 'salir', se detiene el bucle
    if articulo.lower() == 'salir':
        break
    
    # Pedir el precio del artículo y convertirlo a número decimal (float)
    precio = float(input(f"Ingrese el precio de {articulo}: "))
    
    # Guardar el artículo y su precio en el diccionario
    cesta[articulo] = precio

# Mostrar la lista de compra
print("\nLista de la compra")
total = 0  # Variable para acumular el total de la compra
for articulo, precio in cesta.items():
    # Mostrar el artículo y su precio con dos decimales
    print(f"{articulo}: {precio:.2f}")
    total += precio  # Sumar el precio al total

# Mostrar el total de la compra con dos decimales
print(f"\nTotal de la compra: {total:.2f}")
